bluepysnap/morph.py

```diff
@@ -90,11 +107,18 @@ def get(self, node_id, transform=False, extension="swc"):
                 according to `node_id` position in the circuit.
             extension (str): expected filetype extension of the morph file.
         """
-        filepath = self.get_filepath(node_id, extension=extension)
-        result = Morphology(filepath)
+        if extension == "h5":
+            collection = morphio.Collection(self._get_morphology_base(extension))
+            name = self.get_name(node_id)
+            result = collection.load(name, mutable=True)
+        else:
+            filepath = self.get_filepath(node_id, extension=extension)
+            result = morphio.mut.Morphology(filepath)
```

Comment:
I see that `morphio.Collection` can be used to load morphologies in both cases (when the collection is a h5 file, and when it's a directory containing multiple files).
Would the code be simplified if we always use the collection, instead of calling morphio.mut.Morphology?

Reply:
Yes, that would be a possibility. Also, one could factor out the collection constructions and buffer it with `@lru_cache` or so...

Reply:
It's strange, I thought I was already doing that; so I'm not sure why this is an older one?

Let me look through my history.

> Also, one could factor out the collection constructions and buffer it with @lru_cache

is collection creation that expensive?
```
In [2]: %time morphio.Collection('/home/gevaert/src/snap/tests/data/morphologies')
CPU times: user 33 μs, sys: 2 μs, total: 35 μs
Wall time: 36 μs
Out[2]: <morphio._morphio.Collection at 0x7f8c5e9a4f30>
```

Reply:
It would save opening the file and traversing the file system tree to do so. Plus whatever internal accounting HDF5 does. May matter when many thousand processes do it at the same time.

Reply:
I think we can cross that bridge when we get proof that it's a problem; if people are opening morphology files, the bottleneck is loading them, compared to having a Collection, imo.